### PR TITLE
Add CMake to automate build process

### PR DESCRIPTION
**Pull request by kraglalbert**
## Description

Added the CMake `CMakeLists.txt` file to the root of the project, which can be executed by running `cmake .` in the root directory. Also added other required code so that the build works on all 3 main operating systems (OS X, Windows and Lunix).

## Changes proposed

 - Add CMake file so that the project can be built using only one script, independent of the developer's operating system

## How to test it
1. Create a directory named `build` from the root directory
2. `cd build` then run `cmake ..`
3. Check that build succeeded and files were generated in the `build` directory

## Next steps

None.

## Deploy notes

None.

### CODE REVIEW

**Review by AndreiUngur — DISMISSED** [review]
As a first pass, this LGTM.

I'm assuming it's out of the scope for this PR to add a line in the `travis.yml` file [introduced here](https://github.com/scrum-gang/neutralinojs/pull/12/files) to run `cmake .`

**Comment by boggy-d** [comment]
> As a first pass, this LGTM.
> 
> I'm assuming it's out of the scope for this PR to add a line in the `travis.yml` file [introduced here](https://github.com/scrum-gang/neutralinojs/pull/12/files) to run `cmake .`

It is out of scope, yeah. Ideally we would add and run the CMake file with the Dockerfile

**Comment by AndreiUngur** [comment]
> Ideally we would add and run the CMake file with the Dockerfile

Of course, though if it works in all 3 environments (as per the review description) there's no dependency on containerization (or anything else, at the moment) to run this in Travis from what I see.

**Review by boggy-d — COMMENTED** [review]
I think we should add the whole `src` folder to the `.gitignore` seeing it seems it is generated by the `CMakeLists.txt`

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,52 @@
+cmake_minimum_required (VERSION 3.15)
```
Comment (boggy-d) [inline CMakeLists.txt]: is `3.15` really required? It's really a pain to upgrade on ubuntu

```suggestion
cmake_minimum_required (VERSION 3.10)
```

**Comment by erickzhao** [comment]
Is `src` then a good folder name for it? Maybe we should change that. It's confusing.

**Comment by kraglalbert** [comment]
> I think we should add the whole `src` folder to the `.gitignore` seeing it seems it is generated by the `CMakeLists.txt`

That folder was added manually (and isn't generated when running CMake) but I'll add the things that do get generated to the .gitignore

File: cmake-utils/io/fs/CMakeFiles/CMakeDirectoryInformation.cmake [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# CMAKE generated file: DO NOT EDIT!
```
Comment (erickzhao) [inline cmake-utils/io/fs/CMakeFiles/CMakeDirectoryInformation.cmake]: CMAKE generated file

File: cmake-utils/io/fs/CMakeFiles/fs.dir/build.make [hunk 3]
```diff
@@ -0,0 +1,99 @@
+# CMAKE generated file: DO NOT EDIT!
```
Comment (erickzhao) [inline cmake-utils/io/fs/CMakeFiles/fs.dir/build.make]: CMAKE generated file

File: cmake-utils/io/fs/CMakeFiles/fs.dir/flags.make [hunk 4]
```diff
@@ -0,0 +1,10 @@
+# CMAKE generated file: DO NOT EDIT!
```
Comment (erickzhao) [inline cmake-utils/io/fs/CMakeFiles/fs.dir/flags.make]: CMAKE generated file

File: cmake-utils/io/fs/Makefile [hunk 5]
```diff
@@ -0,0 +1,180 @@
+# CMAKE generated file: DO NOT EDIT!
```
Comment (erickzhao) [inline cmake-utils/io/fs/Makefile]: CMAKE generated file

**Review by erickzhao — DISMISSED** [review]
remove generated files

File: cmake-utils/io/fs/CMakeFiles/CMakeDirectoryInformation.cmake (hunk 2 above)
Comment (kraglalbert) [reply cmake-utils/io/fs/CMakeFiles/CMakeDirectoryInformation.cmake]: Good catch! I updated the PR to remove these files

**Review by boggy-d — APPROVED** [review]

**Review by AndreiUngur — APPROVED** [review]

**Pull request merged**